exporter/stackdriverexporter/stackdriver.go

```diff
@@ -109,35 +136,53 @@ func newStackdriverExporter(cfg *Config) (*stackdriver.Exporter, error) {
 		}
 		options.MapResource = rm.mapResource
 	}
-	return stackdriver.NewExporter(options)
+	sde, serr := stackdriver.NewExporter(options)
+	if serr != nil {
+		return nil, fmt.Errorf("cannot configure Stackdriver metric exporter: %w", serr)
+	}
+	mExp := &metricsExporter{mexporter: sde}
+
+	return exporterhelper.NewMetricsExporter(
+		cfg,
+		mExp.pushMetrics,
+		exporterhelper.WithShutdown(mExp.Shutdown))
 }
 
-// pushMetricsData is a wrapper method on StackdriverExporter.PushMetricsProto
-func (se *stackdriverExporter) pushMetricsData(ctx context.Context, md consumerdata.MetricsData) (int, error) {
-	return se.exporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
+// pushMetrics calls StackdriverExporter.PushMetricsProto on each element of the given metrics
+func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) (int, error) {
+	mds := pdatautil.MetricsToMetricsData(m)
+	dropped := 0
+	for _, md := range mds {
+		d, err := me.mexporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
+		dropped += d
+		if err != nil {
+			return dropped, err
+		}
+	}
+	return dropped, nil
 }
 
-// pushTraceData is a wrapper method on StackdriverExporter.PushTraceSpans
-func (se *stackdriverExporter) pushTraceData(ctx context.Context, td consumerdata.TraceData) (int, error) {
+// pushTraces calls texporter.ExportSpan for each span in the given traces
+func (te *traceExporter) pushTraces(ctx context.Context, td pdata.Traces) (int, error) {
```

Comment:
```suggestion
func (te *traceExporter) pushTraces(_ context.Context, td pdata.Traces) (int, error) {
	// NOTE: SDK Trace exporter exports data asynchronously via bundler, 
 	// so we can't use the parent (synchronous) context, which gets cancelled 
	// immediately after this function is executed.
	ctx := context.Background() // create a new context to avoid cancellation
```